vuln-reach/src/javascript/lang/accesses.rs

```diff
@@ -207,6 +215,55 @@ impl<'a> AccessGraph<'a> {
         None
     }
 
+    /// Find the parent scope of a node.
+    fn find_parent_scope(
+        symbol_table: &SymbolTable<'a>,
+        cursor_cache: &mut TreeCursorCache<'a>,
+        node: Node<'a>,
+    ) -> Option<Node<'a>> {
+        let mut parent_cursor = cursor_cache.cursor(node).unwrap();
+        while let Some(node) = parent_cursor.goto_parent() {
+            let parent = parent_cursor.parent()?;
+
+            if parent.child_by_field_name("body") == Some(node)
+                && symbol_table.get_scope(node).is_some()
+            {
+                return Some(node);
+            }
+        }
+        None
+    }
+
+    /// Check where a function is assigned to or executed from.
+    fn find_function_access(
+        symbol_table: &SymbolTable<'a>,
+        cursor_cache: &mut TreeCursorCache<'a>,
+        node: Node<'a>,
+    ) -> Option<Node<'a>> {
+        let mut access_node = None;
+
+        let mut cursor = cursor_cache.cursor(node).unwrap();
+        while let Some(parent) = cursor.goto_parent() {
+            match parent.kind() {
+                // Get identifier if this function is assigned to a variable.
+                "variable_declarator" => access_node = Some(parent),
+                // Find the parent scope for immediately executed anonymous functions.
+                "call_expression" => {
+                    let scope = Self::find_parent_scope(symbol_table, cursor_cache, parent);
+                    access_node = scope?.parent();
+                },
+                // Running into another function without the current one being executed or
+                // assigned means it can never be reached.
+                "function" | "arrow_function" => break,
+                _ => (),
+            }
+        }
+
+        access_node?.child_by_field_name("name")
+            // TODO: Is this actually necessary?
```

Comment:
Access nodes could be `variable_declarator`s whose `name` child could be a [destructuring pattern](https://github.com/tree-sitter/tree-sitter-javascript/blob/5720b249490b3c17245ba772f6be4a43edb4e3b7/grammar.js#L265). In theory, something like

```js
var [foo, bar, baz] = function () { /* ... */ }
```

should be an error, but I'm not confident about edge cases, and chances are the error would only happen at runtime.

Reply:
That's valid JavaScript though, isn't it? Why isn't it handled properly?

If we just ignore it like you did before, we wouldn't correctly compute the reachability, right?

Reply:
That's unhandled only in this context where we are evaluating a parent function though, no? Elsewhere (in the node visitor) it should be handled properly.

`<destructuring assignment> = <function block>` would most likely be an error as a function block can never be an iterator. 

Do we have a better option in this spot?